plugins/inputs/opensearch_query/opensearch_query_test.go

```diff
@@ -0,0 +1,740 @@
+		ExposedPorts: []string{servicePort},
+		Env: map[string]string{
+			"discovery.type": "single-node",
+		},
+		WaitingFor: wait.ForAll(
+			wait.ForLog(".opendistro_security is used as internal security index."),
+			wait.ForListeningPort(nat.Port(servicePort)),
+		),
+	}
+	err = container.Start()
+	require.NoError(t, err, "failed to start container")
+
+	url := fmt.Sprintf("https://%s:%s", container.Address, container.Ports[servicePort])
+
+	o := newOpensearchQuery(url)
+
+	err = o.newClient()
+	if err != nil {
+		return &container, o, err
+	}
+
+	// parse and build query
+	file, err := os.Open("testdata/nginx_logs")
+	if err != nil {
+		return &container, o, err
+	}
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
+
+	indexer, err := opensearchutil.NewBulkIndexer(opensearchutil.BulkIndexerConfig{
+		Client:  o.osClient,
+		Index:   testindex,
+		Refresh: "true",
+	})
+	if err != nil {
+		return &container, o, err
+	}
+
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		parts := strings.Split(scanner.Text(), " ")
+		size, _ := strconv.Atoi(parts[9])
+		responseTime, _ := strconv.Atoi(parts[len(parts)-1])
+
+		logline := nginxlog{
+			IPaddress:    parts[0],
+			Timestamp:    time.Now().UTC(),
+			Method:       strings.ReplaceAll(parts[5], `"`, ""),
+			URI:          parts[6],
+			Httpversion:  strings.ReplaceAll(parts[7], `"`, ""),
+			Response:     parts[8],
+			Size:         float64(size),
+			ResponseTime: float64(responseTime),
+		}
+
+		body, e := json.Marshal(logline)
+		if e != nil {
+			return &container, o, e
+		}
+
+		e = indexer.Add(
+			context.Background(),
+			opensearchutil.BulkIndexerItem{
+				Index:  testindex,
+				Action: "index",
+				Body:   strings.NewReader(string(body)),
+			})
+		if e != nil {
+			return &container, o, e
+		}
+	}
+
+	if scanner.Err() != nil {
+		return &container, o, err
+	}
+
+	if err = indexer.Close(context.Background()); err != nil {
+		return &container, o, err
+	}
+
+	return &container, o, nil
+}
+
+func TestOpensearchQueryIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	container, o, err := setupIntegrationTest(t)
+	require.NoError(t, err)
+	defer container.Terminate()
+
+	for _, tt := range testData() {
+		t.Run(tt.queryName, func(t *testing.T) {
+			var err error
+			var acc testutil.Accumulator
+
+			o.Aggregations = []osAggregation{tt.testAggregationQueryInput}
+			err = o.Init()
+			if (err != nil) != tt.wantInitErr {
+				t.Errorf("OpensearchQuery.Init() error = %v, wantInitErr %v", err, tt.wantInitErr)
+				return
+			} else if err != nil {
+				// Init() failures mean we're done
+				return
+			}
+
+			err = o.Gather(&acc)
+			if (len(acc.Errors) > 0) && !tt.wantQueryResErr {
+				for _, err = range acc.Errors {
+					t.Errorf("OpensearchQuery.Gather() error: %v, wantQueryResErr %v", err, tt.wantQueryResErr)
+				}
+				return
+			}
+
+			require.NoError(t, err)
```

Comment:
Hmm what if we `tt.wantQueryResErr` but do not get any?

Reply:
Easy.  You rewrite the test to properly check.  

`(len(acc.Errors) > 0) != tt.wantQueryResErr` should fix that